packages/plugins/src/hardhat/tasks.ts

```diff
@@ -1352,3 +1355,171 @@ export const listProjectsTask = async ({}, hre: HardhatRuntimeEnvironment) => {
+  const signer = provider.getSigner()
+
+  const spinner = ora({ isSilent: silent })
+  spinner.start('Checking project registration...')
+
+  const parsedConfig = loadParsedChugSplashConfig(configPath)
+
+  // Throw an error if the project has not been registered
+  if (
+    (await isProjectRegistered(signer, parsedConfig.options.projectName)) ===
+    false
+  ) {
+    errorProjectNotRegistered(
+      await getChainId(hre.ethers.provider),
+      hre.network.name,
+      configPath
+    )
+  }
+
+  spinner.succeed('Process registration detected')
+  spinner.start('Checking proxy compatibility...')
+
+  const incompatibleProxyError = `ChugSplash does not support your proxy type.
+  Currently ChugSplash only supports proxies that implement EIP-1967 which yours does not appear to do.
+  If you believe this is a mistake, please reach out to the developers or open an issue on GitHub.`
+
+  // Fetch proxy bytecode and check if it contains the expected EIP-1967 function definitions
+  const iface = new ethers.utils.Interface([
+    'function implementation() external view returns (address)',
+    'function admin() returns (address)',
+    'function upgradeTo(address)',
+    'function changeAdmin(address newAdmin)',
+  ])
+  const bytecode = await provider.getCode(proxyAddress)
+  const checkFunctions = ['implementation', 'admin', 'upgradeTo', 'changeAdmin']
+  for (const func of checkFunctions) {
+    const sigHash = remove0x(iface.getSighash(func))
+    if (!bytecode.includes(sigHash)) {
+      throw new Error(incompatibleProxyError)
+    }
+  }
+
+  // Fetch proxy owner address from storage slot defined by EIP-1967
+  const ownerAddress = BigNumber.from(
+    await provider.getStorageAt(
+      proxyAddress,
+      '0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103'
+    )
+  ).toHexString()
+
+  // Fetch ChugSplashManager address for this project
+  const managerAddress = await getChugSplashManagerProxyAddress(
+    parsedConfig.options.projectName
+  )
+
+  // If proxy owner is not a valid address, then proxy type is incompatible
+  if (!ethers.utils.isAddress(ownerAddress)) {
+    throw new Error(incompatibleProxyError)
+  }
+
+  // If proxy owner is already ChugSplash, then throw an error
+  if (managerAddress.toLowerCase() === ownerAddress.toLowerCase()) {
+    throw new Error('Proxy is already owned by ChugSplash')
+  }
+
+  // If the signer doesn't own the target proxy, then throw an error
+  const signerAddress = await signer.getAddress()
+  if (ownerAddress.toLowerCase() !== signerAddress.toLowerCase()) {
+    throw new Error(`Target proxy is owned by: ${ownerAddress}.
+You attempted to transfer ownership of the proxy using the address: ${signerAddress}`)
+  }
+
+  // Check that the proxy implementation function returns an address
+  const contract = new ethers.Contract(proxyAddress, iface, signer)
+  const implementationAddress = BigNumber.from(
+    await contract.callStatic.implementation()
+  ).toHexString()
```

Comment:
This can just be:
```
const implementationAddress =  await contract.callStatic.implementation()
```
This'll return the 0x-prefixed address without potentially truncating leading zeros.